contracts/FlashRollover.sol

```diff
@@ -0,0 +1,344 @@
+     * flash loan operation. May change based on if the
+     * old loan is on the current loanCore or legacy (in
+     * which case it requires migration).
+     */
+    struct OperationContracts {
+        ILoanCore loanCore;
+        IERC721 borrowerNote;
+        IERC721 lenderNote;
+        IFeeController feeController;
+        IERC721 assetWrapper;
+        IRepaymentController repaymentController;
+        IOriginationController originationController;
+        ILoanCore newLoanLoanCore;
+        IERC721 newLoanBorrowerNote;
+    }
+
+    /* solhint-disable var-name-mixedcase */
+    // AAVE Contracts
+    // Variable names are in upper case to fulfill IFlashLoanReceiver interface
+    ILendingPoolAddressesProvider public immutable override ADDRESSES_PROVIDER;
+    ILendingPool public immutable override LENDING_POOL;
+    /* solhint-enable var-name-mixedcase */
+
+    // Pawn.fi Contracts
+    ILoanCore public immutable loanCore;
+    ILoanCore public immutable legacyLoanCore;
+    IOriginationController public immutable originationController;
+    IRepaymentController public immutable legacyRepaymentController;
+    IRepaymentController public immutable repaymentController;
+    IERC721 public immutable borrowerNote;
+    IERC721 public immutable lenderNote;
+    IERC721 public immutable legacyBorrowerNote;
+    IERC721 public immutable legacyLenderNote;
+    IERC721 public immutable assetWrapper;
+    IFeeController public immutable feeController;
+
+    constructor(
+        ILendingPoolAddressesProvider _addressesProvider,
+        ILoanCore _loanCore,
+        ILoanCore _legacyLoanCore,
+        IOriginationController _originationController,
+        IRepaymentController _repaymentController,
+        IRepaymentController _legacyRepaymentController,
+        IERC721 _borrowerNote,
+        IERC721 _legacyBorrowerNote,
+        IERC721 _lenderNote,
+        IERC721 _legacyLenderNote,
+        IERC721 _assetWrapper,
+        IFeeController _feeController
+    ) {
+        ADDRESSES_PROVIDER = _addressesProvider;
+        LENDING_POOL = ILendingPool(_addressesProvider.getLendingPool());
+        loanCore = _loanCore;
+        legacyLoanCore = _legacyLoanCore;
+        originationController = _originationController;
+        repaymentController = _repaymentController;
+        legacyRepaymentController = _legacyRepaymentController;
+        borrowerNote = _borrowerNote;
+        legacyBorrowerNote = _legacyBorrowerNote;
+        lenderNote = _lenderNote;
+        legacyLenderNote = _legacyLenderNote;
+        assetWrapper = _assetWrapper;
+        feeController = _feeController;
+    }
+
+    function rolloverLoan(
+        bool isLegacy,
+        uint256 loanId,
+        LoanLibrary.LoanTerms calldata newLoanTerms,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) external override {
+        // Get loan details
+        LoanLibrary.LoanData memory loanData;
+        if (isLegacy) {
```

Comment:
I had thought about that, but we still need to pass an `isLegacy` param somehow to `executeOperation`, since AAVE can only ever call one callback function, and it needs to act differently based on if we are migrating.

I figured if there's no avoiding the `isLegacy` parameter there in `executeOperation`, it might as well bubble up to the external interface.